drivers/include/motor_driver.h

```diff
@@ -145,59 +135,81 @@ typedef struct {
     gpio_t gpio_enable;         /**< GPIO to enable/disable motor */
     gpio_t gpio_dir0;           /**< GPIO to control rotation direction */
     gpio_t gpio_dir1_or_brake;  /**< GPIO to control rotation direction */
-    uint8_t gpio_dir_reverse;   /**< flag to reverse direction */
-    uint8_t gpio_enable_invert; /**< flag to set enable GPIO inverted mode */
-    uint8_t gpio_brake_invert;  /**< flag to make brake active low */
+    bool gpio_dir_reverse;      /**< flag to reverse direction */
 } motor_t;
 
 /**
- * @brief   Default motor driver type definition
+ * @brief   Motor driver
  */
-typedef unsigned int motor_driver_t;
+typedef struct _motor_driver_t motor_driver_t;
 
 /**
  * @brief   Motor callback. It is called at end of motor_set()
  */
-typedef void (*motor_driver_cb_t)(const motor_driver_t motor_driver,
+typedef void (*motor_set_post_cb_t)(const motor_driver_t *motor_driver,
                                   uint8_t motor_id,
                                   int32_t pwm_duty_cycle);
 
+/**
+ * @brief   Motor set callback. Called to set motor speed.
+ */
+typedef void (*motor_set_cb_t)(const motor_t *motor,
+                               motor_direction_t direction);
+
+/**
+ * @brief   Motor brake callback. Called to brake a motor.
+ */
+typedef void (*motor_brake_cb_t)(const motor_t *motor,
+                                 bool brake);
+
 /**
  * @brief Describe DC motor driver with PWM device and motors array
  */
 typedef struct {
-    pwm_t pwm_dev;                          /**< PWM device driving motors */
-    motor_driver_mode_t mode;               /**< driver mode */
-    motor_driver_mode_brake_t mode_brake;   /**< driver brake mode */
-    pwm_mode_t pwm_mode;                    /**< PWM mode */
-    uint32_t pwm_frequency;                 /**< PWM device frequency */
-    uint32_t pwm_resolution;                /**< PWM device resolution */
-    uint8_t nb_motors;                      /**< number of moros */
-    motor_t motors[CONFIG_MOTOR_DRIVER_MAX];       /**< motors array */
-    motor_driver_cb_t cb;                   /**< callback on motor_set */
-} motor_driver_config_t;
+    motor_driver_mode_t mode;                   /**< driver mode */
+    pwm_t pwm_dev;                              /**< PWM device driving motors */
+    pwm_mode_t pwm_mode;                        /**< PWM mode */
+    uint32_t pwm_frequency;                     /**< PWM device frequency */
+    uint32_t pwm_resolution;                    /**< PWM device resolution */
+    bool brake_inverted;                        /**< if false, brake high (1), low (0) otherwise */
```

Comment:
```suggestion
    union {
        bool dir1_inverted;                         /**< second direction pin is inverted if true */
        bool brake_inverted;                        /**< if false, brake high (1), low (0) otherwise */
    }
```

and use the value in `_motor_set_two_dirs( `

Reply:
In the case of a motor driver with 2 directions pin, the truth table to drive it is:

```
 __________________________
| DIR0 | DIR1 |  BEHAVIOR  |
|--------------------------|
|  0   |  0   | BRAKE LOW  |
|  0   |  1   |     CW     |
|  1   |  0   |     CCW    |
|  1   |  1   | BRAKE HIGH |
|______|______|____________|
```
So if you invert `gpio_dir1` according to the `dir1_inverted` you propose, it brakes.

`brake_inverted` and `enable_inverted` are here to say if the default brake/enable logical state is high or low. It is defined at the level of the `motor_driver_t` (a driver can drives several motors).

To reverse by default a motor, the `gpio_dir_reverse` is used at the level of `motor_t`. I kept the `_reverse` suffix instead of `_inverted` as I think it has more functional sense in the contet of a motor.